src/extract.cpp

```diff
@@ -245,25 +248,37 @@ void setupTokenTable() {
 
 static thread_local Extractor* extractor;
 
-NAN_METHOD(extractStrings) {
-	Local<Context> context = Nan::GetCurrentContext();
-	int position = Local<Number>::Cast(info[0])->IntegerValue(context).FromJust();
-	int size = Local<Number>::Cast(info[1])->IntegerValue(context).FromJust();
-	if (info[2]->IsArrayBufferView()) {
-		uint8_t* source = (uint8_t*) node::Buffer::Data(info[2]);
-		info.GetReturnValue().Set(extractor->extractStrings(position, size, source));
+Value extractStrings(const CallbackInfo& info) {
+  Env env = info.Env();
+	int position = info[0].As<Number>();
+	int size = info[1].As<Number>();
+	if (info[2].IsTypedArray()) {
+    	void* source = info[2].As<TypedArray>().ArrayBuffer().Data();
+		return extractor->extractStrings(env, position, size, (uint8_t*) source);
 	}
+  return env.Undefined();
 }
 
-NAN_METHOD(isOneByte) {
-	info.GetReturnValue().Set(Nan::New<Boolean>(Local<String>::Cast(info[0])->IsOneByte()));
+Value isOneByte(const CallbackInfo& info) {
+  	Env env = info.Env();
+	size_t length;
+	napi_get_value_string_latin1(env, info[0], nullptr, 0, &length);
+
+  	return Boolean::New(env, length == 1);
```

Comment:
Just FYI, IsOneByte is not a test to see if it is one byte long string, it is a V8 function that tells you if the string is using its internal one-byte-per-character representation, as opposed to a internal UTF16 representation. This function isn't necessary, I had just used it for testing, because it is pretty critical to preserve this optimization in V8 since it is so important memory efficiency and performance. However, I will just delete this function since it really isn't needed anymore (or only use it in a NAN/V8 build).